### PR TITLE
Adds uuid support

### PR DESCRIPTION
**Pull request by jaredculp**
Simple wrapper around the built in python library because I was a bit surprised that the following code didn't work.

```python
from faker import Factory

fake = Factory.create()
fake.uuid()
```

### CODE REVIEW

File: faker/providers/misc/__init__.py [hunk 1]
```diff
@@ -70,6 +71,13 @@ def language_code(cls):
         return cls.random_element(cls.language_codes)
 
     @classmethod
+    def uuid(cls):
```
Comment (fcurella) [inline faker/providers/misc/__init__.py]: can you rename the moethod to be `uuid4`, so it's clear what kind of uuid is returning?

Comment (jaredculp) [inline faker/providers/misc/__init__.py]: Sure! Should I add methods for the other uuid types?

Comment (fcurella) [inline faker/providers/misc/__init__.py]: uuid3 and uuid5 both require some argument, and uuid1 is very insecure, so
I wouldn't bother. But it's really up to you and what you think it's useful
:)

Comment (fcurella) [inline faker/providers/misc/__init__.py]: > Or perhaps just add a parameter to this method, and have it default to uuid4

I'd rather have it in the name method, makes it more explicit.

Comment (jaredculp) [inline faker/providers/misc/__init__.py]: I've only ever used `uuid4` so I think that is sufficient. Do I need to add any tests/docs?

File: faker/providers/misc/__init__.py [hunk 2]
```diff
@@ -70,6 +71,13 @@ def language_code(cls):
         return cls.random_element(cls.language_codes)
 
     @classmethod
+    def uuid4j(cls):
```
Comment (fcurella) [inline faker/providers/misc/__init__.py]: is the `j` a typo?

File: faker/providers/misc/__init__.py (hunk 1 above)
Comment (fcurella) [inline faker/providers/misc/__init__.py]: On Tue, Jun 2, 2015 at 9:56 AM, Jared Culp <notifications@github.com> wrote:

> Do I need to add any tests/docs?


that'd be great, thanks!

**Comment by jaredculp** [comment]
@fcurella I'm not sure what to add for tests/docs so I think this is ready to merge if you're okay with it

**Comment by fcurella** [comment]
Thank you very much! :sparkles:

**Comment by fcurella** [comment]
that's okay, i think the test suite automatically builds a smoke test,
which is pretty much what you would've written anyway.

**Pull request merged**